src/backend/mpi/swdsm.cpp

```diff
@@ -50,16 +53,14 @@ char * pagecopy;
 pthread_mutex_t cachemutex = PTHREAD_MUTEX_INITIALIZER;
 
 /*Writebuffer*/
-/** @brief  Size of the writebuffer*/
-size_t writebuffersize;
-/** @brief  Writebuffer for storing indices to pages*/
-unsigned long  *writebuffer;
-/** @brief Most recent entry in the writebuffer*/
-unsigned long  writebufferstart;
-/** @brief Least recent entry in the writebuffer*/
-unsigned long  writebufferend;
+/** @brief A write buffer storing cache indexes */
+WriteBuffer<unsigned long>* write_buffer;
+/** @brief  Size of the write buffer*/
+size_t write_buffer_size;
+/** @brief  Number of write buffer elements to write back when full */
+size_t write_buffer_write_back_size;
```

Comment:
instead of being global variables, could these two be moved to the WriteBuffer class?

Reply:
Both variables are now part of the WriteBuffer class.